### ISSUE TITLE
ball tracking issue with Multiball & Quick Multiball stack 

### ISSUE THREAD

**Issue by kneissl** [mypinballs/whirlwind#1]
With 2 balls locked for regular multiball I was able to shoot the right ramp enough to light quick multiball. Once the ball was locked in the saucer under the right ramp for a stacked multi ball. The game was then stuck with the ball trough continuously trying to kick out a ball.

**Comment by orange-cloud** [comment]
'Quick multiball lit' option needs disabling if 2 balls locked for regular multiball

**Comment by kneissl** [comment]
Original game would stack them so jackpots were worth an additional million.

**Comment by orange-cloud** [comment]
Interesting. That needs implementing then definitely. So a stack can occur if you get 2 balls locked first, then advance the tornado target up to 'quick multiball' and enter the under ramp saucer. Which music was running when this started? In theory its qm you are still starting but with the additional locked ball.

Currently the code is starting qm play correctly (no bug) but is looking for the next ball in the trough (of course there isn't one), so it needs to be checking the locked ball count first to work out if stacking is achieved an then where to launch the other ball/s from.

I also think a setting of 'enable multiball stacking'  should be added, so that the user can choose whether qm is still possible once 2 balls are locked.

**Comment by kneissl** [comment]
I think the ramp is up when two balls are locked in the original. So, you'd need quick multiball selected from the spinner, then shoot a lit cellar to get the stack in the original. (start around 5:20 explains how it works https://www.youtube.com/watch?v=kNOWXKKNqRw)
With different ramp up/down behavior; Shooting the right ramp to build up to quick multiball with two balls locked could lead to something new/different than the original stack via lit cellar.

**Comment by orange-cloud** [comment]
Yes, I see how williams handled it now. The video is useful, especially to see how the game sets the music and display during these more complicated modes. 

A feature enhancement is needed in the `def lock_ready(self):` method of the `QuickMultiball` class to handle if regular multiball is already 'ready', stop the regular qm lock logic running (and hence any future trough ball kick outs) and set a player stat 'stacked multiball' flag, which can then be used in the mutliball class to raise the jackpot levels.

On future multiballs the ramp is not up at 'multiball ready' You must start from the left ramp, so something could be added here if you advanced the skyway to qm instead of selecting it from the spinner/cellar combination. This would be a separate enhancement
